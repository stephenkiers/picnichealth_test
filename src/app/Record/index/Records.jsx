import React, {Component} from 'react'
import PropTypes from 'prop-types'
import ImmutablePropTypes from 'react-immutable-proptypes'
import RecordContainer from "../show/RecordContainer";


class Records extends Component {

    render() {
        return (
            <div className="index-of-records">
                {
                    this.props.records
                        .sortBy(record => record.get('created_at'))
                        .reverse()
                        .map(record => {
                            return (
                                <RecordContainer
                                    key={record.get('id')}
                                    record={record}
                                />
                            )
                        }).toJS()
                }
            </div>
        )
    }
}
Records.defaultProps = {
};
Records.propTypes = {
    records: ImmutablePropTypes.set.isRequired,
}

export default Records
