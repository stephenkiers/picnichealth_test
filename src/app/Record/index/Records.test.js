import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import Records from './Records';
import {Map, Set} from 'immutable'

test('Records displays correct', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(
        <Records
            records={Set([
                Map({
                    id: "a1b2",
                    first_name: "FirstName",
                    last_name: "LastName",
                    phone_number: "55555555555",
                    created_at: 123456,
                    state: "idle",
                }),
                Map({
                    id: "c3d4",
                    first_name: "FirstName2",
                    last_name: "LastName2",
                    phone_number: "6666666666",
                    created_at: 654321,
                    state: "idle",
                })
            ])}
        />
    );
    const tree = renderer.getRenderOutput()
    expect(tree).toMatchSnapshot();
});