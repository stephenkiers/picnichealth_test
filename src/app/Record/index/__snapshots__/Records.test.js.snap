// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Records displays correct 1`] = `
<div
  className="index-of-records"
>
  <Connect(RecordContainer)
    record={
      Immutable.Map {
        id: "c3d4",
        first_name: "FirstName2",
        last_name: "LastName2",
        phone_number: "6666666666",
        created_at: 654321,
        state: "idle",
      }
    }
  />
  <Connect(RecordContainer)
    record={
      Immutable.Map {
        id: "a1b2",
        first_name: "FirstName",
        last_name: "LastName",
        phone_number: "55555555555",
        created_at: 123456,
        state: "idle",
      }
    }
  />
</div>
`;
